src/pymor/reductors/neural_network.py

```diff
@@ -767,143 +736,23 @@ def _compute_layer_sizes(self, hidden_layers):
-
-                for sample in samples:
-                    self._update_scaling_parameters(sample)
-                self.training_data.extend(samples)
-
-    def _compute_sample(self, mu, output_trajectory=None):
-        """Transform parameter and corresponding output to |NumPy arrays|.
-
-        This function takes care of including the time instances in the inputs.
-        """
-        if output_trajectory:
-            output_size = output_trajectory.shape[0]
+        if self.is_stationary:
+            return [self.fom.parameters.dim, hidden_dimension, len(self.reduced_basis), ]
         else:
-            output_trajectory = self.fom.output(mu)
+            return [self.fom.parameters.dim + 1, hidden_dimension, len(self.reduced_basis), ]
 
-        output_size = output_trajectory.shape[0]
-        samples = [(mu.with_(t=t), output.flatten())
-                   for t, output in zip(np.linspace(0, self.T, output_size), output_trajectory)]
 
-        return samples
-
-    def _compute_layer_sizes(self, hidden_layers):
-        """Compute the number of neurons in the layers of the neural network."""
-        # determine the numbers of neurons in the hidden layers
-        if isinstance(hidden_layers, str):
-            hidden_layers = eval(hidden_layers, {'N': self.dim_output, 'P': self.parameters_dim})
-        # input and output size of the neural network are prescribed by the
-        # dimension of the parameter space and the output dimension
-        assert isinstance(hidden_layers, list)
-        return [self.parameters_dim + 1, ] + hidden_layers + [self.dim_output, ]
-
-    def _build_rom(self):
-        """Construct the reduced order model."""
-        if self.fom:
-            parameters = self.fom.parameters
-            name = self.fom.name
-        else:
-            parameters = self.training_set[0][0].parameters
-            name = 'data_driven'
-
-        with self.logger.block('Building ROM ...'):
-            rom = NeuralNetworkInstationaryStatefreeOutputModel(self.T, self.nt, self.neural_network,
-                                                                parameters=parameters,
-                                                                scaling_parameters=self.scaling_parameters,
-                                                                name=f'{name}_output_reduced')
-
-        return rom
-
-
-class NeuralNetworkLSTMInstationaryStatefreeOutputReductor(NeuralNetworkInstationaryStatefreeOutputReductor,
-                                                           NeuralNetworkLSTMInstationaryReductor):
+class NeuralNetworkLSTMStatefreeOutputReductor(NeuralNetworkStatefreeOutputReductor,
+                                               NeuralNetworkLSTMReductor):
     """Output reductor relying on LSTM neural networks.
 
     This is a reductor that trains an LSTM neural network that approximates
     the mapping from parameter space to output space.
     """
 
```

Comment:
Not sure if these two lines here are needed. I think they are never called, could that be true? If you confirm  we can delete or at least comment them out as well.